pkg/skaffold/initializer/util/util.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/misc"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+// ListBuilders returns a list of builder names being used in the given build config.
+func ListBuilders(build *latest.BuildConfig) []string {
+	if build == nil {
+		return []string{}
+	}
+
+	results := util.NewStringSet()
+	for _, artifact := range build.Artifacts {
+		results.Insert(misc.ArtifactType(artifact))
+	}
+
+	return results.ToList()
+}
+
+// ListDeployers returns a list of deployer names being used in the given deploy config.
+func ListDeployers(deploy *latest.DeployConfig) []string {
```

Comment:
Looks good just thought this function might be useful since I used it for the same use case so I think it could help here too by using it as the underlying function. It just uses the yaml that deploy *latest.DeployConfig unmarshals to, to get the list of deployers.
https://github.com/GoogleContainerTools/skaffold/blob/d781368a5845d71c687e179456cc6ead58e971e8/pkg/skaffold/instrumentation/meter.go#L87
https://github.com/GoogleContainerTools/skaffold/blob/1f200b186c4a28858b39e9f8e689d5606946fca1/pkg/skaffold/yamltags/tags.go#L80-L92

Reply:
Also realizing there's tab right before ```kubectl``` in the comment that shouldn't be there 😅

Reply:
thanks for sharing this, I think your function is probably what I should use here because it should be fine in the case of a new deployer being added to skaffold